identity/identity.go

```diff
@@ -287,3 +301,39 @@ func (i *Identity) ValidateNID() error {
 
 	return nil
 }
+
+// RevealCredential gets params from query and fill identity accordingly
+func (i *Identity) RevealCredential(r *http.Request, c cipher.Provider) error {
+	if r.URL.Query().Get("reveal_credentials") == "oidc_token" {
+		return i.getOIDCToken(r, c)
+	}
+	return nil
+}
+
+func (i *Identity) getOIDCToken(r *http.Request, c cipher.Provider) error {
+	for credType, credential := range i.Credentials {
+		if credType != CredentialsTypeOIDC {
```

Comment:
This is perfect- I just think you don’t need a new struct! Use what we already have with the credentials struct :)